Plugins/Wox.Plugin.ControlPanel/Main.cs

```diff
@@ -4,17 +4,28 @@
 using System.IO;
 using System.Linq;
 using System.Threading.Tasks;
+using System.Windows.Controls;
 using Wox.Infrastructure;
+using Wox.Infrastructure.Storage;
 
 namespace Wox.Plugin.ControlPanel
 {
-    public class Main : IPlugin, IPluginI18n
+    public class Main : IPlugin, IPluginI18n, ISettingProvider, ISavable
     {
+        private readonly Settings _settings = new Settings();
+        private readonly PluginJsonStorage<Settings> _settingsStorage;
```

Comment:
I would declare this as static rather than readonly. readonly will give a different meaning and intent, the settings object can be and often times need to be modified. Particularly if you are implementing the ISavable which communicates the fact that you are saving the settings object after the state is changed.

Reply:
readonly only means that we are only creating the setting object once and I think this is the intent here, static make it on the class level it is a different intention like you said but it make sense here.